src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs

```diff
@@ -323,7 +323,7 @@ static bool hasDefaultScope(bool useUpdatedEscapeRules, AnonymousTypeField field
 
             static bool isValidTypeArgument(bool useUpdatedEscapeRules, AnonymousTypeField field, ref bool needsIndexedName)
             {
-                needsIndexedName = needsIndexedName || field.IsParams || field.DefaultValue is not null;
+                needsIndexedName = needsIndexedName || field.IsParams || field.DefaultValue is not null || field.RefKind == RefKind.RefReadOnlyParameter;
```

Comment:
I didn't understand this change. Why do we need a special case for `ref readonly` when we didn't have a special case for `in`?

Reply:
Good question. That's because `in`/`ref`/`out` modifiers are handled by `RefKindVector` which encodes them into the delegate name (e.g., `void(ref int)` gets name `<>A{00000001}`) - but the new `ref readonly` modifier cannot fit there (there are only two bits per modifier), so I chose to opt it out of this naming scheme and it instead gets unique name like `<>f__AnonymousDelegate1`.

Reply:
What's the purpose of the naming scheme if we exclude various signatures from it?

Reply:
Let's leave a PROTOTYPE comment. A possible alternative approach is to expand the `RefKindVector` to use more bits. So we would just use more digits to encode the name.